sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/LruCache.java

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.core.implementation.util;
+
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.Map;
+import java.util.Queue;
+import java.util.function.Function;
+
+/**
+ * A least recently used cache that will retain the most recently used key-values until a cache size limit is reached.
+ * <p>
+ * The least recently used cache will eject values that have been added or accessed least recently, offering the ability
+ * to limit the size of the cache while still receiving the benefit of retaining the most recently used values. The
+ * cache does have limitations where it is possible for a most commonly used value to be ejected due to a streak of less
+ * commonly but more recently used values being added, which won't be covered in this implementation.
+ */
+public final class LruCache<Key, Value> {
```

Comment:
[nit] Should we name this LRUCache instead? as it is an acronym

Reply:
Our pattern is to use pascal case with acronyms, Lru, Http, Md5, etc

https://azure.github.io/azure-sdk/java_introduction.html#java-naming-uppercase-acronyms